core/amber/src/main/scala/edu/uci/ics/texera/web/TexeraWebApplication.scala

```diff
@@ -223,39 +233,54 @@ class TexeraWebApplication extends io.dropwizard.Application[TexeraWebConfigurat
     * MongoDB doesn't have an API of drop collection where collection name in (from a subquery), so the implementation is to retrieve
     * the entire list of those documents that have expired, then loop the list to drop them one by one
     */
-  def cleanOldCollections(
-      executions: List[ExecutionResultEntry],
-      status: WorkflowAggregatedState
+  def cleanExecutions(
+      executions: List[WorkflowExecutions],
+      statusChangeFunc: Byte => Byte
   ): Unit = {
     // drop the collection and update the status to ABORTED
     executions.foreach(execEntry => {
-      dropCollections(execEntry.result)
+      dropCollections(execEntry.getResult)
+      deleteReplayLog(execEntry.getLogLocation)
       // then delete the pointer from mySQL
-      ExecutionsMetadataPersistService.tryUpdateExecutionStatusAndPointers(
-        execEntry.eId.longValue(),
-        status
-      )
+      ExecutionsMetadataPersistService.tryUpdateExistingExecution(execEntry.getEid.longValue()) {
+        execution =>
+          execution.setResult("")
+          execution.setLogLocation(null)
+          execution.setStatus(statusChangeFunc(execution.getStatus))
+      }
     })
```

Comment:
this clean-up logic is critical. let's have an offline discussion. also, can it be a separated PR since it is quite important logic change by itself.

Reply:
No, it does not make any sense to create a separate PR. The clean-up logic change is related to the creation of log files. All I did was to add the cleanup of logfiles to the logic. The original cleanup logic for result is the same.